src/fights.ts

```diff
@@ -1400,39 +1402,89 @@ const freeFightSources = [
   new FreeFight(
     () => {
       if (!have($item`closed-circuit pay phone`)) return false;
+      // Check if we have or can get Shadow Affinity
       if (have($effect`Shadow Affinity`)) return true;
-      if (get("_shadowAffinityToday")) return false;
+      if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) return true;
 
-      if (!ClosedCircuitPayphone.rufusTarget()) return true;
       if (get("rufusQuestType") === "items") {
         return false; // We deemed it unprofitable to complete the quest in potionSetup
       }
       if (get("encountersUntilSRChoice") === 0) {
         // Target is either an artifact or a boss
         return true; // Get the artifact or kill the boss immediately for free
       }
-      return false; // We have to spend turns to get the artifact or kill the boss
+
+      // Consider forcing noncombats below:
+      if (globalOptions.prefs.yachtzeechain) return false; // NCs are better when yachtzeeing, probably
+      // TODO: With the KoL update, is there a function for checking if an NC is already forced?
+      if (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed) {
```

Comment:
In line 2521 of `fights.ts` we have `if (use($item`Clara's bell`)) return true;` which sets `_claraBellUsed` to `true` without setting `globalOptions.clarasBellClaimed` to `true`. We should fix that too as we expand our reliance on this global option variable.

Reply:
Erm... I'm pretty sure that already uses the global variable. It sets it on line 2520.